### ISSUE TITLE
replace-require-and-define mangles MemberExpressions

### ISSUE THREAD

**Issue by crowjonah** [Esri/ember-cli-amd#54]
In several apps, we're using `ember-cli-amd` to consume the Esri arcgis-js-api. We've recently introduced Stencil components via `calcite-components`, where there's [a known conflict](https://github.com/ArcGIS/calcite-components-examples/issues/3) with this repo.

In https://github.com/Esri/ember-esri-loader/, [they've worked around this issue](https://github.com/Esri/ember-esri-loader/pull/90/files) by replacing the `espree`/`esrima-walk` approach with a regex that explicitly skips `customElements.define`.

I forked this repo and tried to improve the parser approach, but `esprima-walk`'s `walkAddParent` method is broken when also using `range`, and [an open PR there](https://github.com/jrajav/esprima-walk/pull/3) doesn't appear to resolve the issue. Further, that repo hasn't been updated in 4 years.

Long story short, it'd be useful if `ember-cli-amd` could avoid rewriting calls to `customElements.define` at least, but maybe also `*.[require|define]`.

**Comment by ffaubry** [comment]
This has been fixed in version 3.1.0.
Closing.
